client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleOutputGate.java

```diff
@@ -104,11 +104,9 @@ public RemoteShuffleOutputGate(
     this.mapId = shuffleDesc.getShuffleResource().getMapPartitionShuffleDescriptor().getMapId();
     this.attemptId =
         shuffleDesc.getShuffleResource().getMapPartitionShuffleDescriptor().getAttemptId();
-    this.rssMetaServiceHost =
-        ((RemoteShuffleResource) shuffleDesc.getShuffleResource()).getRssMetaServiceHost();
-    this.rssMetaServicePort =
-        ((RemoteShuffleResource) shuffleDesc.getShuffleResource()).getRssMetaServicePort();
-    this.shuffleWriteClient = createWriteClient();
+    this.rssMetaServiceHost = (shuffleDesc.getShuffleResource()).getRssMetaServiceHost();
```

Comment:
We can remove the outer brackets at the same time, and I also suggest we can make the method call shorter than before.

Reply:
Brackets removed. What do you mean by "make the method call shorter than before" ?

Reply:
like
shuffleDesc.getShuffleResource().getRssMetaServiceHost(); -> shuffleResource.getRssMetaServiceHost();
shuffleDesc.getShuffleResource().getRssMetaServicePort(); -> shuffleResource.getRssMetaServicePort();

But this is just a suggestion.

Reply:
Yeah, I also feels that RemoteShuffleDescriptor nests too much, i.e. RemoteShuffleDescriptor.RemoteShuffleResource.ShuffleResourceDescriptor, maybe we can refactor later